scripts/prepare_sector_network.py

```diff
@@ -2229,14 +2229,23 @@ def add_residential(n, costs):
         )
 
         heat_buses = (n.loads_t.p_set.filter(regex="heat").filter(like=country)).columns
-        n.loads_t.p_set.loc[:, heat_buses] = (
-            (
-                n.loads_t.p_set.filter(like=country)[heat_buses]
-                / n.loads_t.p_set.filter(like=country)[heat_buses].sum().sum()
+
+        # Check if the denominator is equal to zero before performing the division
+        denominator = n.loads_t.p_set.filter(like=country)[heat_buses].sum().sum()
+
+        if denominator != 0:
+            # Perform the division only if the denominator is not zero
+            n.loads_t.p_set.loc[:, heat_buses] = (
+                (
+                    n.loads_t.p_set.filter(like=country)[heat_buses]
+                    / n.loads_t.p_set.filter(like=country)[heat_buses].sum().sum()

```

Comment:
I see here the denominator is repeated; we could do "/ denominator" instead.

Personally, I'd love to define a function like:
```
def _safe_division(numerator, denominator):
   return numerator/denominator if denominator != 0.0 else 0.0
```
and then use this fuction without the if/else condition, like:
`n.loads_t.p_set.loc[:, heat_buses] = _safe_division(... , ...)`

What do you think?
That should be quite compact

Reply:
Sounds very usefull! :) Made some adjustments and commited it. What do you think about returning nan instead of 0.0? We could still convert it to 0.0 but maybe in general (except in this case with multiplying afterwards) a good solution? :)